bin/cdn-upload

```diff
@@ -0,0 +1,94 @@
+#!/usr/bin/env node
+process.on('unhandledRejection', err => console.error(err))
+
+const knox = require('knox')
+const cf = require('cloudfront')
+const { promisify } = require('util')
+const { relative } = require('path')
+const semver = require('semver')
+
+const s3Client = knox.createClient({
+    key: process.env.AWS_ACCESS_KEY_ID,
+    secret: process.env.AWS_SECRET_ACCESS_KEY,
+    bucket: 'bugsnagcdn'
+})
+
+const cfClient = cf.createClient(
+    process.env.AWS_ACCESS_KEY_ID,
+    process.env.AWS_SECRET_ACCESS_KEY
+)
+
+// INPUT PARAMETERS
+const packageDir = process.cwd()
+const packageJson = require(packageDir + "/package.json")
+const packageName = packageJson.name
+const packageVersion = packageJson.version
+const invalidationKey = 'E205JDPNKONLN7' // replace with input argument?
+const files = process.argv.slice(2)
+
+const headers = {
+  'Cache-Control': 'public, max-age=315360000',
+  'Access-Control-Allow-Origin': '*',
+  'x-amz-acl': 'public-read'
+}
+
+console.log(`upload files for ${packageName} ${packageVersion}...`)
+
+const putFile = promisify(s3Client.putFile.bind(s3Client))
+
+const upload = async (localPath, remotePath) => {
+  const relativePath = relative('..', localPath)
+  console.log(`uploading ${relativePath} -> ${remotePath}`)
+
+  const response = await putFile(localPath, remotePath, headers)
+
+  // Knox doesn't error if the request fails, so we need to check if this was a successful upload
+  if (response.statusCode >= 400) {
+    throw new Error(`Failed to upload ${relativePath}: ${response.statusCode} ${response.statusMessage}`)
+  }
+
+  return response
+}
+
+const invalidate = (paths) => {
+  console.log(`invalidating CloudFront cache for the following objects:\n${paths.join('\n')}`)
+  return promisify(cfClient.createInvalidation.bind(cfClient))(invalidationKey, `bugsnag-js@${version}`, paths) // TODO: Do we need to get the package name? i.e. bugsnag-js
```

Comment:
Does this invalidation need to be different for the web-worker package?

Reply:
Yeah, I think they need to be different

The "`invalidationKey`" (called "`distributionId`" in AWS docs) will be the same, but the "[CallerReference](https://docs.aws.amazon.com/cloudfront/latest/APIReference/API_CreateInvalidation.html#API_CreateInvalidation_RequestSyntax)" [should be different](https://docs.aws.amazon.com/cloudfront/latest/APIReference/API_CreateInvalidation.html#API_CreateInvalidation_RequestSyntax):

> CloudFront uses the value to prevent you from accidentally resubmitting an identical request. Whenever you create a new invalidation request, you must specify a new value for CallerReference and change other values in the request as applicable
> &hellip;
> If CallerReference is a value you already sent in a previous invalidation batch request but the content of any Path is different from the original request, CloudFront returns an InvalidationBatchAlreadyExists error.

It's completely arbitrary though, so using the package name + version should be fine

```suggestion
  return promisify(cfClient.createInvalidation.bind(cfClient))(invalidationKey, `${packageName}@${packageVersion}`, paths)
```